### ISSUE TITLE
during inference, how to handle unlabeled class?

### ISSUE THREAD

**Issue by wzhouuu** [davidtvs/PyTorch-ENet#2]
Hi, thank your for your great work. I've successfully trained my own models on local dataset. Just one question about unlabeled class: based on my understanding, the unlabeled class is excluded by setting the weight to 0 during training. But during inference, I found the trained model couldn't predict any unlabeled class. It predicted a cyan boundary (like the car class in attached image) where cyan color supposed to be pole and sign-symbols..Not sure if there's anything I've done wrong? 
![results](https://user-images.githubusercontent.com/24283641/39412683-917eb208-4c63-11e8-979d-84d45a6a3017.png)

**Comment by davidtvs** [comment]
Sorry for the late reply. Somehow I never noticed your issue until now.

Yes, your understanding is correct, the unlabeled class is ignored by setting its weight to 0. Does the same problem happen with training images? i.e. with a trained model, if you run inference on the training images do you see the same issues?

**Comment by wzhouuu** [comment]
yep, it seems to be the same..I tried two version, one is to set unlabeled class to have weight 0, the other one is include unlabeled class during training.. but both seem to have this problem..I couldn't come up with any idea why this happened because my pole class is 2, sign class is 6, they are not either first number or the last number...

**Comment by davidtvs** [comment]
After looking into this a bit more, seems that you are right.
![figure_1](https://user-images.githubusercontent.com/17404144/40881584-2e21d888-66c2-11e8-8458-33682805b310.png)
The top row of images are ground-truth labels and the bottom row are the predictions. The model is indeed making predictions for unlabeled pixels.

**Comment by davidtvs** [comment]
Apparently, this behaviour is expected. After looking at the author's implementation of ENet and other segmentation networks my conclusion is that there is no way to stop the network from predicting a class for the unlabeled pixels. Those pixels are simply ignored when computing metrics, but a prediction is always made.

I have pushed a few changes to the IoU metric that add an additional argument (`ignore_index`) which allows for a class to be ignored when computing metrics.

If you want unlabeled pixels to be predicted as unlabeled, use the flag `--with-unlabeled`. @wzhouuu make sure you are using the current version of this repository as the old flag (`--ignore_unlabeled´) had no effect.

**Comment by wzhouuu** [comment]
Thanks for the update!
